tests/test_atoms_state.py

```diff
@@ -0,0 +1,350 @@
+#
+# Copyright The NOMAD Authors.
+#
+# This file is part of NOMAD. See https://nomad-lab.eu for further info.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import pytest
+import numpy as np
+
+from nomad.units import ureg
+
+from . import logger
+
+from nomad_simulations.atoms_state import (
+    OrbitalsState,
+    CoreHole,
+    HubbardInteractions,
+    AtomsState,
+)
+
+
+class TestOrbitalsState:
+    """
+    Test the `OrbitalsState` class defined in atoms_state.py.
+    """
+
+    @staticmethod
+    def add_state(
+        orbital_state, l_number, ml_number, ms_number, j_number, mj_number
+    ) -> None:
+        """Adds l and ml quantum numbers to the `OrbitalsState` section."""
+        orbital_state.l_quantum_number = l_number
+        orbital_state.ml_quantum_number = ml_number
+        orbital_state.ms_quantum_number = ms_number
+        orbital_state.j_quantum_number = j_number
+        orbital_state.mj_quantum_number = mj_number
+
+    @pytest.mark.parametrize(
+        'number, values, results',
+        [
+            ('n_quantum_number', [-1, 0, 1, 2], [False, False, True, True]),
+            ('l_quantum_number', [-2, 0, 1, 2], [False, False, True, True]),
+            # l_quantum_number == 2 when testing 'ml_quantum_number'
+            ('ml_quantum_number', [-3, 5, -2, 1], [False, False, True, True]),
+            ('ms_quantum_number', [0, 10, -0.5, 0.5], [False, False, True, True]),
+        ],
+    )
```

Comment:
> Question: what about states that contain multiple electrons?

Here I am only defining orbital states, not the electronic one. I think the electronic could be define a posteriori using some normalization, that's why I open a discussion #22, so that we can safely define electronic state **from** `OrbitalsState`